File: cpp/Longest_Balanced_Substring_I_3713.cpp
class Solution {
public:
    int longestBalanced(string s) {
        int maxWindow = 0;
        for (int i = 0; i < s.length(); i++) {
            vector <int> frequency(26);
            for (int j = i; j < s.length(); j++) {
                frequency[s[j] - 'a']++;
                int minCount = 1e9, maxCount = 0;
                for (int k = 0; k < 26; k++) {
                    if (frequency[k] > 0) {
                        minCount = min(minCount, frequency[k]);
                        maxCount = max(maxCount, frequency[k]);
                    }
                }
                if (minCount == maxCount) {
                    maxWindow = max(maxWindow, j - i + 1);
                }
            }
        }
        return maxWindow;
    }
};

File: python/Longest_Balanced_Substring_I_3713.py
class Solution:
    def longestBalanced(self, s: str) -> int:
        maxWindow = 0
        for i in range(len(s)):
            frequency = [0] * 26
            for j in range(i, len(s)):
                frequency[ord(s[j]) - ord('a')] += 1
                minCount, maxCount = 1e9, 0
                for freq in frequency:
                    if freq > 0:
                        minCount = min(minCount, freq)
                        maxCount = max(maxCount, freq)
                if minCount == maxCount:
                    maxWindow = max(maxWindow, j - i + 1)
        return maxWindow
